### ISSUE TITLE
Checking for variables fails with interaction terms

### ISSUE THREAD

**Issue by anniejw6** [anniejw6/modmarg#53]
```
data(mtcars)
mod <- glm(mpg ~ cyl * poly(disp, degree = 2, raw = T),
           data = mtcars, family = gaussian)
z <- modmarg::mod_marg2(
  mod = mod, var_interest = 'cyl', type = 'effects',
  at = list('disp' = c(140, 160))
)
```
Gives me an error of
```
Error in `[.data.frame`(mod$data, , all.vars(mod$formula)) : 
  undefined columns selected
```

Can we also add this as a test?

**Comment by anniejw6** [comment]
```
> all.vars(mod$formula, functions = F)
[1] "mpg"  "cyl"  "disp" "T"   
> all.vars(mod$formula, functions = F, unique = T)
[1] "mpg"  "cyl"  "disp" "T"   
> mod <- glm(mpg ~ cyl * poly(disp, degree = 2, raw = TRUE),
+            data = mtcars, family = gaussian)
> all.vars(mod$formula, functions = F, unique = T)
[1] "mpg"  "cyl"  "disp"
```

So... sort of gets me there.

**Comment by anniejw6** [comment]
Resolved in #54
